File: tests/cpp/test_geometry.cpp
#include <catch2/catch_test_macros.hpp>
#include "../../src/cpp/modules/geometry/geometry_manager.hpp"

TEST_CASE("Geometry grid initialization", "[Geometry]") {
  auto wafer = std::make_shared<Wafer>(300.0, 775.0, "silicon");
  GeometryManager geometry;
  geometry.initializeGrid(wafer, 10, 10);
  REQUIRE(wafer->getGrid().rows() == 10);
  REQUIRE(wafer->getGrid().cols() == 10);
}

TEST_CASE("Geometry apply layer", "[Geometry]") {
  auto wafer = std::make_shared<Wafer>(300.0, 775.0, "silicon");
  GeometryManager geometry;
  geometry.applyLayer(wafer, 0.1, "oxide");
  REQUIRE(wafer->getThickness() == 775.1);
  REQUIRE(wafer->getMaterialId() == "oxide");
}

File: tests/python/test_geometry.py
import pytest
from semiconductor_simulator.geometry import PyGeometryManager, PyWafer

def test_geometry_initialize_grid():
    wafer = PyWafer(300.0, 775.0, "silicon")
    geometry = PyGeometryManager()
    geometry.initialize_grid(wafer, 10, 10)
    # Add grid access in PyWafer for testing if needed

def test_geometry_apply_layer():
    wafer = PyWafer(300.0, 775.0, "silicon")
    geometry = PyGeometryManager()
    geometry.apply_layer(wafer, 0.1, "oxide")
    # Add material/thickness access for testing